deploy/runtools/runtime_config.py

```diff
@@ -295,42 +323,26 @@ def __init__(self, runtimeconfigfile: str, configoverridedata: str) -> None:
-
-        self.always_expand = runtime_dict['run_farm'].get('always_expand_runfarm', "yes") == "yes"
-
-        # Default values
-        self.trace_enable = False
-        self.trace_select = "0"
-        self.trace_start = "0"
-        self.trace_end = "-1"
-        self.trace_output_format = "0"
-        self.autocounter_readrate = 0
-        self.zerooutdram = False
-        self.disable_asserts = False
-        self.print_start = "0"
-        self.print_end = "-1"
-        self.print_cycle_prefix = True
-
+        self.tracerv_config = TracerVConfig()
         if 'tracing' in runtime_dict:
-            self.trace_enable = runtime_dict['tracing'].get('enable') == "yes"
-            self.trace_select = runtime_dict['tracing'].get('selector', "0")
-            self.trace_start = runtime_dict['tracing'].get('start', "0")
-            self.trace_end = runtime_dict['tracing'].get('end', "-1")
-            self.trace_output_format = runtime_dict['tracing'].get('output_format', "0")
+            self.tracerv_config.enable = runtime_dict['tracing'].get('enable') == "yes"
+            self.tracerv_config.select = runtime_dict['tracing'].get('selector', "0")
+            self.tracerv_config.start = runtime_dict['tracing'].get('start', "0")
+            self.tracerv_config.end = runtime_dict['tracing'].get('end', "-1")
+            self.tracerv_config.output_format = runtime_dict['tracing'].get('output_format', "0")
+        self.autocounter_config = AutoCounterConfig()
         if 'autocounter' in runtime_dict:
-            self.autocounter_readrate = int(runtime_dict['autocounter'].get('read_rate', "0"))
+            self.autocounter_config.readrate = int(runtime_dict['autocounter'].get('read_rate', "0"))
         self.defaulthwconfig = runtime_dict['target_config']['default_hw_config']
+        self.hostdebug_config = HostDebugConfig()
         if 'host_debug' in runtime_dict:
-            self.zerooutdram = runtime_dict['host_debug'].get('zero_out_dram') == "yes"
-            self.disable_asserts = runtime_dict['host_debug'].get('disable_synth_asserts') == "yes"
+            self.hostdebug_config.zero_out_dram = runtime_dict['host_debug'].get('zero_out_dram') == "yes"
+            self.hostdebug_config.disable_synth_asserts = runtime_dict['host_debug'].get('disable_synth_asserts') == "yes"
+        self.synthprint_config = SynthPrintConfig()
         if 'synth_print' in runtime_dict:
-            self.print_start = runtime_dict['synth_print'].get("start", "0")
-            self.print_end = runtime_dict['synth_print'].get("end", "-1")
-            self.print_cycle_prefix = runtime_dict['synth_print'].get("cycle_prefix", "yes") == "yes"
+            self.synthprint_config.start = runtime_dict['synth_print'].get("start", "0")
+            self.synthprint_config.end = runtime_dict['synth_print'].get("end", "-1")
+            self.synthprint_config.cycle_prefix = runtime_dict['synth_print'].get("cycle_prefix", "yes") == "yes"
```

Comment:
nit. In a future PR, it would probably be good to push the initialization of these configs into their respective `__init__` methods.